src/contract/assembled_transaction.ts

```diff
@@ -754,11 +755,22 @@ export class AssembledTransaction<T> {
                 entry.credentials().address().signature().switch().name ===
                 "scvVoid"),
           )
-          .map((entry) =>
-            StrKey.encodeEd25519PublicKey(
-              entry.credentials().address().address().accountId().ed25519(),
-            ),
-          ),
+          .map((entry) => {
+            const addr = entry.credentials().address().address();
+            const addrType = addr.switch().name;
+            switch(addrType) {
+              case 'scAddressTypeAccount':
```

Comment:
nit: using the value directly is a more reliable approach in case these changes.

you can see how it's done in [scValToNative](https://stellar.github.io/js-stellar-sdk/js-stellar-base_src_scval.js.html#line296), something like

```suggestion
            const addrType = addr.switch().value;
            switch (addrType) {
              case xdr.ScAddressType.scAddressTypeAccount().value:
```

Reply:
I know it's longer and more verbose but it's also faster since it's a digit check instead of a string :shrug: more idiomatic for sure, anyhow